src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -289,6 +290,8 @@ public TopicPartitionChannel(
             this.offsetPersistedInSnowflake,
             this.processedOffset,
             this.latestConsumerOffset);
+    this.telemetryServiceV2.reportKafkaPartitionStart(
+        new SnowflakeTelemetryChannelCreation(this.tableName, this.channelName));
```

Comment:
Is this the only place that we're creating channel? If no, we need to either cover all cases or update the metric name to something like partition start

Reply:
Yes this is the only place we create a channel. There are two constructors for TopicPartitionChannel. The test constructor calls into this constructor, so all TopicPartitionChannel object creations will go through this codepath

Reply:
I thought we also reopen/create channel when the channel gets invalidated? Not sure if this is the same code path though

Reply:
The TopicPartitionChannel object does not get recreated when the channel is invalidated, so the current reportKafkaPartitionStart tracks when a new TopicPartitionChannel is created.

Alternatively we could emit a reportKafkaPartitionStart with each channel open, however I don't think this is a good idea because we can track this server side and we already have multiple channel open/close issues that may overload the telemetry service with events. 

There are two code paths for opening channels: [TopicPartitionChannel constructor](https://github.com/snowflakedb/snowflake-kafka-connector/blob/bc7af5e54774697d297eb31b4a6980c98cca7588/src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java#L277) and when we [recover offset from snowflake](https://github.com/snowflakedb/snowflake-kafka-connector/blob/bc7af5e54774697d297eb31b4a6980c98cca7588/src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java#L977) (channel invalidation).

I will add the a channel open count to the reportPartitionUsage event as a way to track the sequencer